django_declarative_apis/machinery/tasks.py

```diff
@@ -183,7 +184,21 @@ def schedule_future_task_runner(task_runner_args, task_runner_kwargs,
         countdown=countdown
     )
     task_runner_kwargs['correlation_id'] = _get_correlation_id()
-    future_task_runner.apply_async(task_runner_args, task_runner_kwargs, queue=queue, routing_key=routing_key, countdown=countdown+delay)
+
+    MAX_ATTEMPTS = 3
+    for attempt in range(MAX_ATTEMPTS):
+        # XXX: This is an attempt to skirt around an unsolved, low repro issue somewhere in the celery/kombu/redis-py stack.
+        # Once in a while, a connection in the pool will timeout prior to a health check being called in redis-py and
+        # will result in an error being raised here. This should be removed once the issue has been sorted out.
+        # Note: This is around the use of redis-py in celery where celery's event loop is not running
+        # https://github.com/celery/kombu/issues/1019
+        try:
+            future_task_runner.apply_async(task_runner_args, task_runner_kwargs, queue=queue, routing_key=routing_key, countdown=countdown+delay)
+            return
+        except kombu.exceptions.OperationalError as err:
+            logger.warn('kombu.exceptions.OperationalError (attempt: %s)', attempt)
+            if attempt >= MAX_ATTEMPTS - 1:
+                raise err
```

Comment:
I'd like to have the option to fall back to running the task synchronously, perhaps controlled by a django setting, or else an option in the `DeferrableTask` decorator